runtime/lua/vim/lsp/health.lua

```diff
@@ -51,6 +51,24 @@ end
 
 local function check_watcher()
   vim.health.start('vim.lsp: File watcher')
+
+  local clients = vim.lsp.get_clients()
+  if
+    --- @param client vim.lsp.Client
+    vim.iter(clients):all(function(client)
+      local has_capability = vim.tbl_get(
+        client.capabilities,
+        'workspace',
+        'didChangeWatchedFiles',
+        'dynamicRegistration'
+      )
+      return has_capability == nil or client.workspace_folders == nil
+    end)
+  then
+    report_info('No clients support file watching')
```

Comment:
(not a blocker) should we tend to mention the literal LSP name?
```suggestion
    report_info('No clients support file watching (workspace/didChangeWatchedFiles)')
```

Reply:
Or `file watching (workspace/didChangeWatchedFiles) disabled on all clients`

(To imply that users can opt-in, and enable it)